src/zappy_server_src/core/command_hold.c

```diff
@@ -0,0 +1,176 @@
+        return true;
+    return false;
+}
+
+/// \brief Cross all the ai command and try to match a command
+/// \param command Command info to fill
+/// \param cmd Command to match
+/// \return true When command is found
+/// \return false When command is not found
+static bool find_ai_command(command_data_t *command, char *cmd)
+{
+    int pos = 0;
+
+    for (; ai_command_list[pos].name; pos++)
+        if (!strncmp(ai_command_list[pos].name, cmd,
+        strlen(ai_command_list[pos].name)))
+            break;
+    if (!ai_command_list[pos].name)
+        return false;
+    command->name = ai_command_list[pos].name;
+    command->ptr = ai_command_list[pos].ptr;
+    if (cmd[strlen(ai_command_list[pos].name)] == ' ' &&
+    ai_command_list[pos].arg) {
+        command->arg = cmd + strlen(ai_command_list[pos].name) + 1;
+        return true;
+    }
+    command->arg = NULL;
+    if (cmd[strlen(ai_command_list[pos].name)] == '\0' &&
+    !ai_command_list[pos].arg)
+        return true;
+    return false;
+}
+
+/// \brief Cross all the unknown command and try to match a command
+/// \param command Command info to fill
+/// \param cmd Command to match
+/// \return true When command is found
+/// \return false When command is not found
+static bool find_unknown_command(command_data_t *command, char *cmd)
+{
+    int pos = 0;
+
+    for (; unknown_command_list[pos].name; pos++)
+        if (!strncmp(unknown_command_list[pos].name, cmd,
+        strlen(unknown_command_list[pos].name)))
+            break;
+    if (!unknown_command_list[pos].name)
+        return false;
+    command->name = unknown_command_list[pos].name;
+    command->ptr = unknown_command_list[pos].ptr;
+    if (cmd[strlen(unknown_command_list[pos].name)] == ' ' &&
+    unknown_command_list[pos].arg) {
+        command->arg = cmd + strlen(unknown_command_list[pos].name) + 1;
+        return true;
+    }
+    command->arg = NULL;
+    if (cmd[strlen(unknown_command_list[pos].name)] == '\0' &&
+    !unknown_command_list[pos].arg)
+        return true;
+    return false;
+}
+
+/// \brief Parse and match the given command to a existing one
+/// \param command Command to parse
+/// \param player_info Player informations
+/// \return command_data_t* Newly created command data
+static command_data_t *find_command_data(char *command,
+player_list_t *player_info)
+{
+    command_data_t *command_data = malloc(sizeof(command_data_t));
+    bool operation = false;
+
+    if (command_data == NULL)
+        return NULL;
+    if (player_info->type == GUI)
+        operation = find_gui_command(command_data, command);
+    if (player_info->type == AI)
+        operation = find_ai_command(command_data, command);
+    if (player_info->type == UNKNOWN)
+        operation = find_unknown_command(command_data, command);
+    if (operation == false) {
+        free(command_data);
+        return NULL;
+    }
+    return command_data;
+}
+
+void compute_command(char *command, player_list_t *player_info,
```

Comment:
Missing Documentation

Reply:
The documentation of a function is in a .h actually in command_hold.h